docs/modules/architecture/pages/vector-search-overview.adoc

```diff
@@ -0,0 +1,52 @@
+= Vector Database Engine
```

Comment:
```suggestion
= VectorCollection data structure design
```

I would avoid the term database because it creates expectations that our AP infrastructure doesn't supply.